src/main/java/jcifs/internal/smb2/Smb2ErrorContextResponse.java

```diff
@@ -0,0 +1,146 @@
+        // for the SMB2 ERROR Context structure
+        int bufferIndex = 0;
+
+        // ErrorDataLength (4 bytes)
+        this.errorDataLengthLength = SMBUtil.readInt4(buffer, bufferIndex);
+        bufferIndex += 4;
+
+        // ErrorId (4 bytes)
+        // STATUS_STOPPED_ON_SYMLINK is always 0x00000000
+        this.errorId = SMBUtil.readInt4(buffer, bufferIndex);
+        if ( this.errorId != 0 ) {
+            throw new SMBProtocolDecodingException("ErrorId should be 0 for STATUS_STOPPED_ON_SYMLINK");
+        }
+        else {
+            bufferIndex += 4;
+            return this.readSymLinkErrorResponse( buffer, bufferIndex );
+        }
+    }
+
+    /**
+     * 2.2.2.2.1 Symbolic Link Error Response
+     * 
+     * @param buffer
+     * @param bufferIndex
+     * @return The length, in bytes, of the response including the variable-length portion
+     * and excluding SymLinkLength
+     * @throws Smb2ProtocolDecodingException
+     */
+    protected int readSymLinkErrorResponse ( byte[] buffer, int bufferIndex ) throws SMBProtocolDecodingException {
+        // SymLinkLength (4 bytes)
+        int symLinkLength = SMBUtil.readInt4(buffer, bufferIndex);
+        bufferIndex += 4;
+
+        // SymLinkErrorTag (4 bytes) (always 0x4C4D5953)
+        int symLinkErrorTag = SMBUtil.readInt4(buffer, bufferIndex);
+        if ( !Integer.toHexString(symLinkErrorTag).toUpperCase().equals("4C4D5953") ) {
+            throw new SMBProtocolDecodingException("SymLinkErrorTag should be 0x4C4D5953");
+        }
+        bufferIndex += 4;
+
+        // skip, not needed
+        bufferIndex += 4; // ReparseTag (4 bytes) (always 0xA000000C)
+        bufferIndex += 2; // ReparseDataLength (2 bytes)
+
+        // UnparsedPathLength (2 bytes)
+        int unparsedPathLength = SMBUtil.readInt2(buffer, bufferIndex);
+        bufferIndex += 2;
+
+        // SubstituteNameOffset (2 bytes)
+        int substituteNameOffset = SMBUtil.readInt2(buffer, bufferIndex);
+        bufferIndex += 2;
+
+        // SubstituteNameLength (2 bytes)
+        int substituteNameLength = SMBUtil.readInt2(buffer, bufferIndex);
+        bufferIndex += 2;
+
+        // PrintNameOffset (2 bytes)
+        int printNameOffset = SMBUtil.readInt2(buffer, bufferIndex);
+        bufferIndex += 2;
+
+        // PrintNameLength (2 bytes)
+        int printNameLength = SMBUtil.readInt2(buffer, bufferIndex);
+        bufferIndex += 2;
+
+        // Flags (4 bytes)
+        this.absolutePath = SMBUtil.readInt4(buffer, bufferIndex) == 0;
+        bufferIndex += 4;
+
+        // PathBuffer (variable), substitute name
+        this.substituteName = Strings.fromUNIBytes(buffer, substituteNameOffset + bufferIndex, substituteNameLength);
+
+        // PathBuffer (variable), print name that also identifies the target of the symbolic link
+        this.printName = Strings.fromUNIBytes(buffer, printNameOffset + bufferIndex, printNameLength);
+
+        return symLinkLength;
```

Comment:
Again this will be complex logic and IMHO I don't believe it brings much value here, however if you have a suggested way of doing it then I'm all for it.